### PR TITLE
Updates after testing

### PR DESCRIPTION
**Pull request by gmafrafortuna**
changes to reduce files left at the end of the run and some rules updates.

### CODE REVIEW

File: Snakemake/config/tsinfer_cattle.yaml [hunk 1]
```diff
@@ -1,45 +0,0 @@
...
-  24 : 62317253
-  25 : 42350435
-  26 : 51992305
-  27 : 45612108
-  28 : 45940150
-  29 : 51098607
-  X : 139009144
-  Y : 43300181
-  M : 16340
-
```
Comment (janaobsteter) [inline Snakemake/config/tsinfer_cattle.yaml:45]: Why is this file gone?

File: Snakemake/workflow/Snakefile [hunk 2]
```diff
@@ -39,6 +43,9 @@ vcfdir=config['vcfDir'] # from config file, the path to vcf
 Project = config['PROJECT']
 mapdir = config['genmap']
 
+# if want to split multiallelic sites set to TRUE:
```
Comment (janaobsteter) [inline Snakemake/workflow/Snakefile:46]: What does this mean?

File: Snakemake/workflow/profile/cluster.json [hunk 3]
```diff
@@ -13,8 +13,8 @@
  },
  "phase":
   {
-    "mem": "150G",
-    "threads": "20",
+    "mem": "350G",
```
Comment (janaobsteter) [inline Snakemake/workflow/profile/cluster.json:16]: In here, just make sure you have a "G" next to the number for mem - otherwise it will complain

File: Snakemake/workflow/profile/config.yaml [hunk 4]
```diff
@@ -2,11 +2,11 @@ restart-times: 0
...
-#drmaa: " -P roslin_HighlanderLab -cwd -l h_vmem={cluster.mem} -l h_rt={cluster.time} -pe sharedmem {cluster.threads}"
+#jobname: "{rule}.{jobid}"
+#drmaa: " -cwd -l h_vmem={cluster.mem} -l h_rt={cluster.time} -pe sharedmem {cluster.threads}"
 rerun-incomplete: true
 rerun-triggers: mtime
 cores: 99 # how many jobs you want to submit to your cluster queue
 local-cores: 1
 keep-going: false
-
+#-P roslin_HighlanderLab
```
Comment (janaobsteter) [inline Snakemake/workflow/profile/config.yaml:12]: Why did you remove this?

File: Snakemake/workflow/rules/merge_samples_vcf.smk [hunk 5]
```diff
@@ -14,34 +14,61 @@ if len(allFiles) != 1:
...
+                            shell('bcftools view -S ^{params.duplicated} {file2}.vcf.gz -O z -o {file2}.filtered.vcf.gz') 
+                            shell('bcftools index {file2}.filtered.vcf.gz')
+                            shell('rm {params.duplicated}')
+                                           
+            for file in input:
+                if file not in filtered:
+                    file1 = file.split('.')[0]
+                    print(f'file {file1} not filtered')
+                    shell("cp {file1}.vcf.gz {file1}.filtered.vcf.gz && bcftools index {file1}.filtered.vcf.gz")
+            shell('rm {params.duplicated}')
```
Comment (janaobsteter) [inline Snakemake/workflow/rules/merge_samples_vcf.smk:64]: What is this bit?

File: Snakemake/workflow/rules/phasing.smk [hunk 6]
```diff
@@ -19,18 +19,21 @@ else:
         params: 
             map = f'{mapdir}/{{chromosome}}.txt',
         #log: 'logs/phase_{chromosome}.log'
-        threads: 1
+        threads: 25
```
Comment (janaobsteter) [inline Snakemake/workflow/rules/phasing.smk:22]: I think this number should be put in the config

File: Snakemake/workflow/rules/prepare_files_for_tsinfer.smk [hunk 7]
```diff
@@ -48,7 +48,12 @@ rule extract_vcf_pos:
     input: rules.get_major.output
         #rules.decompress.output
     output:
+<<<<<<< HEAD
```
Comment (janaobsteter) [inline Snakemake/workflow/rules/prepare_files_for_tsinfer.smk:51]: There are some cofnlicts here, that is why you have this "HEAD". This needs to be cleaned before merging

File: Snakemake/workflow/rules/prepare_files_for_tsinfer.smk [hunk 8]
```diff
@@ -66,19 +71,28 @@ rule extract_vcf_pos:
 
 rule match_ancestral_vcf:
     input:
+<<<<<<< HEAD
+        vcfPos = rules.extract_vcf_pos.output.file,
+        ancestral = "AncestralAllele/AncestralAllele_Vcf.txt",
+        major = rules.get_major.output
+    output: 
+        file = temp(f'{vcfdir}/AncestralVcfMatch{{chromosome}}.txt'),
+=======
```
Comment (janaobsteter) [inline Snakemake/workflow/rules/prepare_files_for_tsinfer.smk:80]: Same here

File: Snakemake/workflow/rules/prepare_files_for_tsinfer.smk [hunk 9]
```diff
@@ -66,19 +71,28 @@ rule extract_vcf_pos:
...
         vcfPos=rules.extract_vcf_pos.output,
         ancestral="AncestralAllele/AncestralAllele_Vcf.txt",
         major=rules.get_major.output
     output: temp('AncestralVcfMatch{chromosome}.txt')
+>>>>>>> 27fdb4deb7416d7d35cf7bb91ea51acbeeb1a66a
     params:
         chrNum = lambda wc: wc.get("chromosome")[3:],
-        ancestral_sites = '{chromosome}.aa'
+        ancestral_sites = f'{vcfdir}/{{chromosome}}.aa',
+        tmp_file = f'{vcfdir}/{{chromosome}}.tmp',
```
Comment (janaobsteter) [inline Snakemake/workflow/rules/prepare_files_for_tsinfer.smk:89]: Why do we need this temp_file?

File: Snakemake/workflow/rules/split.smk [hunk 10]
```diff
@@ -1,12 +1,15 @@
...
     for f in os.listdir(f'{vcfdir}/RawVCF')
-        if (f.endswith("vcf.gz") and f.startswith("Chr"))]))
+        if (f.endswith("vcf.gz") and (f.startswith("Chr") | f.startswith("chr")) )]))
 combinedFiles = list(set([f.strip("vcf.gz").split("_")[1]
     for f in os.listdir(f'{vcfdir}/RawVCF')
         if (f.endswith("vcf.gz") and f.startswith("Combined"))]))
 allFiles = splitFiles + combinedFiles
 
+#print(splitFiles)
+#print(combinedFiles)
```
Comment (janaobsteter) [inline Snakemake/workflow/rules/split.smk:11]: Can you clean these comments?

File: Snakemake/workflow/scripts/PrepareTsinferSampleFile.py [hunk 11]
```diff
@@ -120,10 +120,10 @@ def add_populations(vcf, samples, metaData):
     Return: A list of population indexes
     """
     pop_lookup = {}
-    metaPop = metaData[['Pop', 'SubPop']].drop_duplicates().dropna(axis = 0, how = 'all')
+    metaPop = metaData.iloc[:, 1:].drop_duplicates().dropna(axis = 0, how = 'all')
     sample_subpop = [list(metaData.SubPop[metaData.ID == x]) for x in vcf.samples]
-    for subpop, pop in zip(metaPop.SubPop, metaPop.Pop):
-        pop_lookup[subpop] = samples.add_population(metadata={"pop": pop, "subpop": subpop})
+    for subpop, pop, origin, lat, long in zip(metaPop.SubPop, metaPop.Pop, metaPop.Origin, metaPop.Lat, metaPop.Long):
```
Comment (janaobsteter) [inline Snakemake/workflow/scripts/PrepareTsinferSampleFile.py:125]: Is this gonna be the standard meta data? To include lat and long? Can these be missing values?

File: Snakemake/workflow/scripts/PrepareTsinferSampleFile.py [hunk 12]
```diff
@@ -120,10 +120,10 @@ def add_populations(vcf, samples, metaData):
...
+        pop_lookup[subpop] = samples.add_population(metadata={"pop": pop, "subpop": subpop, "country": origin, "coord": f'{lat}/{long}'})
     return [pop_lookup[subpop[0]] for subpop in sample_subpop]
```
Comment (janaobsteter) [inline Snakemake/workflow/scripts/PrepareTsinferSampleFile.py:127]: What happens if lat/long are missing?

**Review by janaobsteter — COMMENTED** [review]
@gmafrafortuna I looked at this - it's mostly questions for me to understand a bit better what is hapenning

File: Snakemake/config/tsinfer_cattle.yaml (hunk 1 above)
Comment (gmafrafortuna) [reply Snakemake/config/tsinfer_cattle.yaml:45]: because this was the one for my run. Don't need to track here (we have a general one already from the tutorial)

File: Snakemake/workflow/Snakefile (hunk 2 above)
Comment (gmafrafortuna) [reply Snakemake/workflow/Snakefile:46]: I am testing something. I've updated now and comment that out

File: Snakemake/workflow/profile/config.yaml (hunk 4 above)
Comment (gmafrafortuna) [reply Snakemake/workflow/profile/config.yaml:12]: I am not sure if that work on the wwn. But I'll put it back because we are only submitting on normal nodes anyway.

File: Snakemake/workflow/rules/merge_samples_vcf.smk (hunk 5 above)
Comment (gmafrafortuna) [reply Snakemake/workflow/rules/merge_samples_vcf.smk:64]: This is just to rename files that haven't been filtered - so they all have the same standard name

File: Snakemake/workflow/rules/prepare_files_for_tsinfer.smk (hunk 9 above)
Comment (gmafrafortuna) [reply Snakemake/workflow/rules/prepare_files_for_tsinfer.smk:82]: This is to allocate the output of the for loop. At the end it is converted into the output. 
(I've just renamed the tmp files so there's no conflict when running multiple chromosomes at once. I actually don't think we need this tmp file because what is being done with the last awk command after the loop can be done with the sed command cemented out, if I understand correctly what is happening. )

File: Snakemake/workflow/scripts/PrepareTsinferSampleFile.py (hunk 11 above)
Comment (gmafrafortuna) [reply Snakemake/workflow/scripts/PrepareTsinferSampleFile.py:125]: It will work as long as the columns are there, even if they are empty. 
I want to make this optional but haven't figured out yet how

File: Snakemake/workflow/scripts/PrepareTsinferSampleFile.py (hunk 12 above)
Comment (gmafrafortuna) [reply Snakemake/workflow/scripts/PrepareTsinferSampleFile.py:127]: if the columns are missing, it will break.

File: Snakemake/workflow/rules/phasing.smk (hunk 6 above)
Comment (gmafrafortuna) [reply Snakemake/workflow/rules/phasing.smk:22]: I'm just using as a variable.

File: Snakemake/workflow/Snakefile (hunk 2 above)
Comment (janaobsteter) [reply Snakemake/workflow/Snakefile:46]: @gmafrafortuna there is still some "HEAD" lines in here

File: Snakemake/workflow/profile/cluster.json [hunk 13]
```diff
@@ -13,8 +13,8 @@
...
+    "threads": "25",
```
Comment (janaobsteter) [inline Snakemake/workflow/profile/cluster.json:17]: Here, you need to change to 25G

File: Snakemake/workflow/profile/cluster.json [hunk 14]
```diff
@@ -44,4 +44,4 @@
     "threads": "20",
```
Comment (janaobsteter) [inline Snakemake/workflow/profile/cluster.json:44]: Here, you need to change to 350G

File: Snakemake/workflow/profile/config.yaml (hunk 4 above)
Comment (janaobsteter) [reply Snakemake/workflow/profile/config.yaml:12]: I am using this on normal nodes so I think it would be good if it's in there :)

File: Snakemake/workflow/rules/merge_samples_vcf.smk (hunk 5 above)
Comment (janaobsteter) [reply Snakemake/workflow/rules/merge_samples_vcf.smk:64]: is this the "same" rule as before, just different name of the rule?

File: Snakemake/workflow/rules/prepare_files_for_tsinfer.smk [hunk 15]
```diff
@@ -64,21 +63,27 @@ rule extract_vcf_pos:
     # bcftools query -f '%CHROM %POS\n' {input} > {output.sites}
     #     awk '{{print $1"_"$2}}' {output.sites} > {output.file}
 
+if config['ancestralAllele'] not null:
```
Comment (janaobsteter) [inline Snakemake/workflow/rules/prepare_files_for_tsinfer.smk:66]: I think this one should not be neccessary? We have somewhere in the code that if it is null, then it runs the AA bit

File: Snakemake/workflow/rules/prepare_files_for_tsinfer.smk (hunk 9 above)
Comment (janaobsteter) [reply Snakemake/workflow/rules/prepare_files_for_tsinfer.smk:82]: Yeah, I think we couldn't have done it without, but it's just an intermediate file. I think it would be better to move to to the "output" instead of "params" and mark it as tmp()

File: Snakemake/workflow/scripts/PrepareTsinferSampleFile.py (hunk 12 above)
Comment (janaobsteter) [reply Snakemake/workflow/scripts/PrepareTsinferSampleFile.py:127]: Did you remove this now?

**Review by janaobsteter — COMMENTED** [review]
@gmafrafortuna , I had a look - I think this isn't your last push because there is still some git conflicts in there. I have some minor questions left :)

File: Snakemake/workflow/profile/cluster.json (hunk 13 above)
Comment (gregorgorjanc) [reply Snakemake/workflow/profile/cluster.json:17]: Is this 25, the number of threads or RAM per thread?

File: Snakemake/workflow/Snakefile (hunk 2 above)
Comment (gmafrafortuna) [reply Snakemake/workflow/Snakefile:41]: fixed
